src/http/http_client_curl.cpp

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright (c) 2022 Snowplow Analytics Ltd. All rights reserved.
+
+This program is licensed to you under the Apache License Version 2.0,
+and you may not use this file except in compliance with the Apache License Version 2.0.
+You may obtain a copy of the Apache License Version 2.0 at http://www.apache.org/licenses/LICENSE-2.0.
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the Apache License Version 2.0 is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the Apache License Version 2.0 for the specific language governing permissions and limitations there under.
+*/
+
+#if !defined(WIN32) && !defined(_WIN32) && !defined(__WIN32) || defined(__CYGWIN__)
+#include "http_client_curl.hpp"
+#include "../constants.hpp"
+#include "curl/curl.h"
+
+using namespace snowplow;
+using std::cerr;
+using std::endl;
+using std::lock_guard;
+
+HttpClientCurl::HttpClientCurl() {
+  curl_global_init(CURL_GLOBAL_ALL);
+}
+
+HttpClientCurl::~HttpClientCurl() {
+  curl_global_cleanup();
+}
+
+const string HttpClientCurl::TRACKER_AGENT = string("Snowplow C++ Tracker (Unix)");
+
+static size_t write_data(void *data, size_t byte_size, size_t n_bytes, std::string *body) {
+  return byte_size * n_bytes;
+}
+
+HttpRequestResult HttpClientCurl::http_request(const RequestMethod method, CrackedUrl url, const string &query_string, const string &post_data, list<int> row_ids, bool oversize) {
+  CURL *curl = curl_easy_init();
+
+  if (curl) {
```

Comment:
Just a very nit opinion: maybe we can return the failure option early?

Reply:
Makes sense, changed!